### PR TITLE
(SIMP-3445) Add acceptance test for consul configuration and bootstrap

### PR DESCRIPTION
**Pull request by brandonrdn**
Add acceptance test for consul configuration and bootstrap. This acceptance test needs to check both libkv::consul::server = true and libkv::consul::server = false, and test the bootstrap run with puppet apply using the bootstrap/consul.pp file.
We need to make sure that after running puppet apply bootstrap/consul.pp, then a puppet agent run including the class, that consul is configured correctly

SIMP-3445 #comment closed

### CODE REVIEW

File: manifests/consul.pp [hunk 1]
```diff
@@ -4,52 +4,62 @@
...
-  $version = '0.8.5',
-  $use_puppet_pki = true,
-  $bootstrap = false,
-  $dont_copy_files = false,
-  $serverhost = undef,
-  $http_listen = '127.0.0.1',
-  $https_listen = '0.0.0.0',
-  $advertise = undef,
-  $datacenter = undef,
+  $firewall          = false,
```
Comment (trevor-vaughan) [inline manifests/consul.pp]: Tie this to the `simp_options` catalyst per the SIMP puppet-module-skeleton: https://github.com/simp/puppet-module-skeleton/blob/master/skeleton/manifests/init.pp.erb#L58

File: manifests/consul.pp [hunk 2]
```diff
@@ -4,52 +4,62 @@
...
+  $server            = false,
+  $version           = '0.8.5',
+  $use_puppet_pki    = true,
```
Comment (trevor-vaughan) [inline manifests/consul.pp]: Have this match the puppet-module-skeleton https://github.com/simp/puppet-module-skeleton/blob/master/skeleton/manifests/init.pp.erb#L56

File: bootstrap/consul.pp [hunk 3]
```diff
@@ -1,38 +1,38 @@
 file { "/etc/simp":
-	ensure => directory,
+  ensure => directory,
+}
```
Comment (trevor-vaughan) [inline bootstrap/consul.pp]: I'm 90% sure that this is going to conflict with the `simp` module itself.

I get that it's a bootstrap class, but it should be able to run with everything else without issue in general. We might need to change both to an 'ensure_resource' or move them up to a higher level class but dragging along the 'simp' module really shouldn't be a big deal in any environment. It's well namespaced and we're using the materials here anyway.

Comment (heliocentric) [reply bootstrap/consul.pp]: We shouldn't bring in the simp module for libkv; it's intended to also stand completely alone. I think ensure_resource for that directory is the best bet

**Comment by jeefberkey** [comment]
Needs rebasing

**Comment by trevor-vaughan** [comment]
@heliocentric @brandonrdn It was indicated that this could not have the parameters pulling from simp_options (with the default `false` as per usual), could you explain further?

**Comment by heliocentric** [comment]
yea; simplib needs to pull this module in as a dependency, so this module can't be dependent on simplib

**Comment by trevor-vaughan** [comment]
@heliocentric OK, that makes sense. Will we need to set these parameters explicitly in Hiera via `simp config` to make all of this work properly?

**Pull request merged**